src/storage/storage.cc

```diff
@@ -103,6 +103,11 @@ void Storage::SetWriteOptions(const Config::RocksDB::WriteOptions &config) {
   write_opts_.memtable_insert_hint_per_batch = config.memtable_insert_hint_per_batch;
 }
 
+void Storage::SetReadOptions(rocksdb::ReadOptions &read_options) {
```

Comment:
It sounds like the `SetReadOptions` function sets the specified `read_options` to the whole `Storage` but the function doesn't do that. For me, the prior name of the function wasn't ideal but definitely more clear :)
The code you are trying to eliminate is repetitive but it's tricky to do that.
I recommend passing `read_options` by a pointer to stress that this parameter will be changed and can be considered as output.

Reply:
According to the official document, mutil_get will also improve the performance. I think we can add this flag to all readoptions

Reply:
I meant that the `Storage` class stores the `write_options` variable as its member, but the `ReadOptions` is passed to the iterator's constructor.

Reply:
Yes, maybe `SetSeekOptions` is more precise.